libr/io/p/io_r2k.c

```diff
@@ -65,7 +65,7 @@ static int r2k__close(RIODesc *fd) {
 	}
 #elif defined (__linux__) && !defined (__GNU__)
 	if (fd) {
-		close (fd->fd);
+		close ((int)fd->data);
```

Comment:
Tried not to touch it more as needed. size_t is specified to be at least 16-bit the same foes for int